sys/include/puf_sram.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_puf_sram SRAM PUF
+ * @ingroup     sys
+ *
+ * @brief       SRAM based physically unclonable function (PUF)
+ *
+ * @{
+ * @file
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ */
+#ifndef PUF_SRAM_H
+#define PUF_SRAM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "hashes.h"
+#include "thread.h"
+
+/**
+ * @brief SRAM length considered for seeding
+ */
+#ifndef SEED_RAM_LEN
+#define SEED_RAM_LEN     (2048 / sizeof(uint32_t))
+#endif
+
+/**
+ * @brief SRAM marker to detect reboot without power-off
+ *
+ * Source: https://www.random.org/bytes/
+ */
+#define PUF_SRAM_MARKER  (0xad3021ff)
+
+/**
+ * @brief Global seed variable, allocated on the stack in puf_sram.c
+ */
+extern uint32_t puf_sram_seed;
+
+/**
+ * @brief Global seed state, allocated on the stack in puf_sram.c
+ */
+extern uint32_t puf_sram_state;
+
+/**
+ * @brief builds hash from @p SEED_RAM_LEN bytes uninitialized SRAM, writs it
+ *        to the global variable @p puf_sram_seed and returns the value
+ *
+ * @param[in] ram pointer to SRAM memory
+ *
+ * @return  a random number
+ */
+uint32_t puf_sram_uint32(const uint8_t *ram);
```

Comment:
It seems unintuitive that this function takes a pointer, but not a length. I suggest making either both the pointer and the length into globals, or both as arguments.
The name is a bit misleading, when comparing with random_uint32 which returns a random uint32_t value. How about puf_sram_generate()?